packages/taiko-client/pkg/state_indexer/indexer.go

```diff
@@ -153,20 +160,36 @@ func (s *Indexer) fetchHistoricalProposals(toBlock *types.Header, bufferSize uin
 			startHeight = new(big.Int).Sub(currentHeader.Number, new(big.Int).SetUint64(maxBlocksPerFilter))
 		}
 
-		log.Info("Fetching Shasta Proposed events", "from", startHeight, "to", currentHeader.Number)
+		log.Info("Fetching Shasta events (Proposed + Proved)", "from", startHeight, "to", currentHeader.Number)
 
-		iter, err := eventiterator.NewBatchProposedIterator(s.ctx, &eventiterator.BatchProposedIteratorConfig{
-			RpcClient:             s.rpc,
-			MaxBlocksReadPerEpoch: &maxBlocksPerFilter,
-			StartHeight:           startHeight,
-			EndHeight:             currentHeader.Number,
-			OnBatchProposedEvent:  s.onProposedEvent,
-		})
+		logs, err := s.ethGetLogsCombined(s.ctx, startHeight, currentHeader.Number)
```

Comment:
shall we further optimize it by querying logs (e.g. 64 * 500 blocks) in parallel (64 requests in parallel)? We might end up making some extra requests for collecting enough events, but in terms of overall time savings, seems it still feels worthwhile?